manager/controllers/app/moduleinstance.go

```diff
@@ -213,123 +213,236 @@ func (m *ModuleManager) selectCopyModule(item modules.DataInfo, appContext *app.
 	return copySelector, nil
 }
 
-// SelectModuleInstances selects the necessary read/copy/write modules for the blueprint for a given data set
+// SelectModuleInstances selects the necessary read/copy/write modules for the plotter for a given data set
+// Adds the asset details, flows and templates to the given plotter spec.
 // Write path is not yet implemented
-func (m *ModuleManager) SelectModuleInstances(item modules.DataInfo, appContext *app.FybrikApplication) ([]modules.ModuleInstanceSpec, error) {
+func (m *ModuleManager) AddFlowInfoForAsset(item modules.DataInfo, appContext *app.FybrikApplication, plotterSpec *app.PlotterSpec, flowType app.DataFlow) error {
 	datasetID := item.Context.DataSetID
 	m.Log.Info("Select modules for " + datasetID)
-	instances := make([]modules.ModuleInstanceSpec, 0)
+
+	subflows := make([]app.SubFlow, 0)
+	assets := map[string]app.AssetDetails{}
+	templates := []app.Template{}
 	var err error
 	if m.WorkloadGeography, err = m.GetProcessingGeography(appContext); err != nil {
```

Comment:
I'm wondering if this should really be an error, or perhaps just a warning, or whether it's appropriate to check here at all.  If there are no rules in the policy manager about geography, then in theory we wouldn't need this.  Right?

Reply:
In the future this function will be applied in a different place, at the earlier stage.

Reply:
opened #812